phantazm-neuron/src/main/java/com/github/phantazmnetwork/neuron/node/Node.java

```diff
@@ -0,0 +1,311 @@
+     * @param h the new h-score
+     */
+    public void setH(float h) {
+        this.h = h;
+    }
+
+    /**
+     * Determines if this node is currently present inside a {@link NodeQueue} by checking its heap index.
+     * @return true if this node is present in a NodeQueue, false otherwise
+     */
+    boolean isOnHeap() {
+        return heapIndex > -1;
+    }
+
+    /**
+     * Sets the heap index. This is only used internally by {@link NodeQueue}.
+     * @param heapIndex the new heap index, or -1 to indicate removal
+     */
+    void setHeapIndex(int heapIndex) {
+        this.heapIndex = heapIndex;
+    }
+
+    /**
+     * Gets the current heap index.
+     * @return the current heap index, or a negative value if this node is not part of any {@link NodeQueue}
+     */
+    int getHeapIndex() {
+        return heapIndex;
+    }
+
+    /**
+     * Gets the parent node. May be null if this node has no parent.
+     * @return the parent node
+     */
+    public @Nullable Node getParent() {
+        return parent;
+    }
+
+    /**
+     * Sets the parent of this node, which may be null.
+     * @param parent the new parent
+     */
+    public void setParent(@Nullable Node parent) {
+        this.parent = parent;
+    }
+
+    /**
+     * <p>Reverses the ordering of the linked list represented by this node. That is, given the following structure of
+     * different nodes labelled a, b, and c, with arrows pointing to the parent of the node on the left:</p>
+     *
+     * <code>
+     *     a -> b -> c
+     * </code>
+     *
+     * <p>Calling reverse will produce the following list:</p>
+     *
+     * <code>
+     *     c -> b -> a
+     * </code>
+     *
+     * <p>The returned node will be the last node in the list before it was inverted. In the example shown above, this
+     * is {@code c}.</p>
+     *
+     * <p>Special case: when this node has no parent, this node is returned.</p>
+     * @return the starting node of the reversed linked list, which was the last node in the list before reversal
+     */
+    public @NotNull Node reverse() {
+        Node prev = null;
+        for(Node node : this) {
+            node.setParent(prev);
+            prev = node;
+        }
+
+        //prev will never be null
+        return Objects.requireNonNull(prev);
+    }
+
+    /**
+     * Gets the position of this node.
+     * @return a {@link Vec3I} instance representing the position of this node
+     */
+    public @NotNull Vec3I getPosition() {
+        return position;
+    }
+
+    /**
+     * Creates a new random-access list implementation with the same nodes as represented by this one.
+     * @return a random-access List containing the same nodes as this
+     */
+    public @NotNull List<Node> toRandomAccessList() {
+        ArrayList<Node> nodes = new ArrayList<>();
+        for(Node node : this) {
+            nodes.add(node);
+        }
+
+        nodes.trimToSize();
+        return nodes;
+    }
```

Comment:
is this method really necessary

Reply:
not at this point, thought it might be useful for debugging, but it's not used anywhere